Code and parm modification for the RRFS-CMAQ chemistry calciulation

hsinmulin-NOAA opened the pull request.

hsinmulin-NOAA committed:
New modification for the RRFS-CMAQ PMTF (PM2.5)

New modification for the RRFS-CMAQ

hsinmulin-NOAA committed:
Modification to include necessary variables for the RRFS-CMAQ

Modification to include necessary variables for the RRFS-CMAQ

WenMeng-NOAA commented on parm/fv3lam_cmaq.xml-pmtf_only:
```diff
@@ -0,0 +1,3423 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
@hsinmulin-NOAA Please remove all of your testing files like this one from your PR.

hsinmulin-NOAA replied:
removed

WenMeng-NOAA commented on parm/fv3lam_cmaq.xml-org:
```diff
@@ -0,0 +1,3953 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
@hsinmulin-NOAA This file should be removed from your PR.

hsinmulin-NOAA replied:
removed

WenMeng-NOAA commented on parm/makefile:
```diff
@@ -45,6 +46,8 @@ GEFSANLCTRLFILENAME     = postcntrl_gefs_anl.xml
 GEFSCTRLF00FILENAME     = postcntrl_gefs_f00.xml
 FV3LAMAVAILXMLFILENAME  = post_avblflds.xml
 FV3LAMCTRLFILENAME      = fv3lam.xml
+FV3LAMCMAQAVAILXMLFILENAME  = fv3lam_cmaq_post_avblflds.xml
```
@hsinmulin-NOAA The avblflds.xml was unfiled. All new variables should be added in post_avblflds.xml.

hsinmulin-NOAA replied:
no additional vars are added, just separated CMAQ use to others

hsinmulin-NOAA replied:
The change is based on the originally existed "fv3lam_post_avblflds.xml", "fv3lam.xml" and "makefile" structure. new "fv3lam_cmaq_post_avblflds.xml" & "fv3lam_cmaq.xml" and the modified "makefile" is to avoid the variables used in RRFS-CMAQ as dummy variables in LAM.  It will create "postxconfig-NT-fv3lam_cmaq.txt" specifically used for RRFS-CMAQ model output

hsinmulin-NOAA replied:
Updated.
Using unified "post_avblflds.xml"  per removal of "fv3lam_cmaq_post_avblflds.xml".

WenMeng-NOAA commented on parm/makefile-cmaq:
```diff
@@ -0,0 +1,154 @@
+SHELL=/bin/sh
```
This file should be removed.

hsinmulin-NOAA replied:
removed

WenMeng-NOAA commented on parm/makefile-org:
```diff
@@ -0,0 +1,147 @@
+SHELL=/bin/sh
```
This file should be removed.

hsinmulin-NOAA replied:
removed

WenMeng-NOAA commented on sorc/ncep_post.fd/ALLOCATE_ALL.f-org:
```diff
@@ -0,0 +1,1295 @@
+!> @file
```
This file should be removed.

hsinmulin-NOAA replied:
removed

WenMeng-NOAA commented on sorc/ncep_post.fd/VRBLS3D_mod.f-org:
```diff
@@ -0,0 +1,84 @@
+!   01-10-22  H CHUANG - MODIFIED TO PROCESS HYBRID MODEL OUTPUT
```
This file should removed.

hsinmulin-NOAA replied:
removed

WenMeng-NOAA commented:
@hsinmulin-NOAA Please open a new issue for this PR. Did you conduct the UPP tests?

hsinmulin-NOAA committed:
Delete fv3lam_cmaq_post_avblflds.xml-org

hsinmulin-NOAA committed:
Delete fv3lam_cmaq.xml-org

hsinmulin-NOAA committed:
Delete fv3lam_cmaq.xml-pmtf_only

hsinmulin-NOAA committed:
Delete fv3lam_cmaq_post_avblflds.xml-new

hsinmulin-NOAA committed:
Delete makefile-cmaq

hsinmulin-NOAA committed:
Delete makefile-org

hsinmulin-NOAA committed:
Delete ALLOCATE_ALL.f-org

hsinmulin-NOAA committed:
Delete ALLOCATE_ALL.f-pmtf

hsinmulin-NOAA committed:
Delete DEALLOCATE.f-org

hsinmulin-NOAA committed:
Delete DEALLOCATE.f-pmtf

hsinmulin-NOAA committed:
Delete INITPOST_NETCDF.f-org

hsinmulin-NOAA committed:
Delete INITPOST_NETCDF.f-pmtf

hsinmulin-NOAA committed:
Delete MDLFLD.f-org

hsinmulin-NOAA committed:
Delete MDLFLD.f-pmtf

hsinmulin-NOAA committed:
Delete MDLFLD.f-pmtf-unit

hsinmulin-NOAA committed:
Delete RQSTFLD.F-org

hsinmulin-NOAA committed:
Delete RQSTFLD.F-pmtf

hsinmulin-NOAA committed:
Delete VRBLS3D_mod.f-org

hsinmulin-NOAA committed:
Delete VRBLS3D_mod.f-pmtf

hsinmulin-NOAA commented:
UPP regression test has been done and passed.
The log file can be seen at VENUS
/gpfs/dell2/emc/modeling/noscrub/Hsin-Mu.Lin/upp-rg/rt.log.WCOSS_DELL_P3

Also below
07/14 12:33:46Z -Starting nmmb test
07/14 12:33:48Z -Starting fv3gefs test
07/14 12:33:50Z -Starting fv3r test
07/14 12:34:06Z -fv3gefs test: your new post executable generates bit-identical gefs.t00z.master.grb2f384 as the trunk
07/14 12:34:06Z -Ending fv3gefs test
07/14 12:34:30Z -nmmb test: your new post executable generates bit-identical BGDAWP03.tm00.Grib2 as the trunk
07/14 12:34:30Z -nmmb test: your new post executable generates bit-identical BGRD3D03.tm00.Grib2 as the trunk
07/14 12:34:31Z -nmmb test: your new post executable generates bit-identical BGRDSF03.tm00.Grib2 as the trunk
07/14 12:34:31Z -Ending nmmb test
07/14 12:34:42Z -Starting nmmb pe test
07/14 12:34:54Z -Runtime: nmmb_test 53 -- baseline 49
07/14 12:35:15Z -nmmb pe test: your new post executable generates bit-identical BGDAWP03.tm00.Grib2 as the trunk
07/14 12:35:16Z -nmmb pe test: your new post executable generates bit-identical BGRD3D03.tm00.Grib2 as the trunk
07/14 12:35:16Z -nmmb pe test: your new post executable generates bit-identical BGRDSF03.tm00.Grib2 as the trunk
07/14 12:35:16Z -Ending nmmb pe test
07/14 12:35:27Z -Starting fv3gefs test
07/14 12:35:35Z -Runtime: nmmb_pe_test 43 -- baseline 37
07/14 12:35:43Z -fv3gefs pe test: your new post executable generates bit-identical gefs.t00z.master.grb2f384 as the trunk
07/14 12:35:43Z -Ending fv3gefs pe test
07/14 12:35:45Z -Runtime: fv3gefs_test 26 -- baseline 40
07/14 12:35:55Z -Starting fv3gfs test
07/14 12:36:11Z -Runtime: fv3gefs_pe_test 25 -- baseline 40
07/14 12:36:47Z -fv3r test: your new post executable generates bit-identical PRSLEV06.tm00 as the trunk
07/14 12:36:50Z -fv3r test: your new post executable generates bit-identical NATLEV06.tm00 as the trunk
07/14 12:36:50Z -Ending fv3r test
07/14 12:37:01Z -Starting fv3r pe test
07/14 12:40:06Z -fv3r pe test: your new post executable generates bit-identical PRSLEV06.tm00 as the trunk
07/14 12:40:09Z -fv3r pe test: your new post executable generates bit-identical NATLEV06.tm00 as the trunk
07/14 12:40:09Z -Ending fv3r pe test
07/14 12:50:41Z gfs_post_00.53876-fv3gfs test: your new post executable generates bit-identical gfs.t00z.master.grb2f006 as the trunk
07/14 12:50:42Z gfs_post_00.53876-fv3gfs test: your new post executable generates bit-identical gfs.t00z.sfluxgrbf006.grib2 as the trunk
07/14 12:50:42Z gfs_post_00.53876-fv3gfs test: your new post executable generates bit-identical gfs.t00z.special.grb2f006 as the trunk
07/14 12:50:42Z gfs_post_00.53876-Ending fv3gfs test
07/14 12:50:54Z -Starting fv3gfs pe test
07/14 12:51:08Z -Runtime: fv3gfs_test 898 -- baseline 1100
07/14 13:06:30Z gfs_post_00.60563-fv3gfs pe test: your new post executable generates bit-identical gfs.t00z.master.grb2f006 as the trunk
07/14 13:06:31Z gfs_post_00.60563-fv3gfs pe test: your new post executable generates bit-identical gfs.t00z.sfluxgrbf006.grib2 as the trunk
07/14 13:06:31Z gfs_post_00.60563-fv3gfs pe test: your new post executable generates bit-identical gfs.t00z.special.grb2f006 as the trunk
07/14 13:06:31Z gfs_post_00.60563-Ending fv3gfs pe test
07/14 13:06:42Z -Starting hrrr test
07/14 13:06:51Z -Runtime: fv3gfs_pe_test 948 -- baseline 1100
07/14 13:07:02Z -Runtime: fv3r_test 189 -- baseline 570
07/14 13:07:12Z -Runtime: fv3r_pe_test 198 -- baseline 644
07/14 13:09:53Z -hrrr test: your new post executable generates bit-identical WRFTWO.GrbF04 as the trunk
07/14 13:09:54Z -hrrr test: your new post executable generates bit-identical WRFPRS.GrbF04 as the trunk
07/14 13:09:55Z -hrrr test: your new post executable generates bit-identical WRFNAT.GrbF04 as the trunk
07/14 13:09:55Z -Ending hrrr test
07/14 13:10:06Z -Starting hrrr pe test
07/14 13:10:23Z -Runtime: hrrr_test 203 -- baseline 100
07/14 13:13:07Z -hrrr pe test: your new post executable is fine in WRFTWO.GrbF04
07/14 13:13:07Z -hrrr pe test: your new post executable generates bit-identical WRFPRS.GrbF04 as the trunk
07/14 13:13:09Z -hrrr pe test: your new post executable generates bit-identical WRFNAT.GrbF04 as the trunk
07/14 13:13:09Z -Ending hrrr pe test
07/14 13:13:19Z -Starting rap test
07/14 13:13:34Z -Runtime: hrrr_pe_test 192 -- baseline 100
07/14 13:14:37Z -rap test: your new post executable generates bit-identical WRFPRS.GrbF16 as the trunk
07/14 13:14:38Z -rap test: your new post executable generates bit-identical WRFNAT.GrbF16 as the trunk
07/14 13:14:38Z -Ending rap test
07/14 13:14:50Z -Starting rap pe test
07/14 13:15:01Z -Runtime: rap_test 88 -- baseline 100
07/14 13:16:10Z -rap pe test: your new post executable is fine in WRFPRS.GrbF16
07/14 13:16:11Z -rap pe test: your new post executable generates bit-identical WRFNAT.GrbF16 as the trunk
07/14 13:16:11Z -Ending rap pe test
07/14 13:16:41Z -Runtime: rap_pe_test 91 -- baseline 100

hsinmulin-NOAA commented:
Extra codes are removed and necessary parms are explained in the comments.

Regression test is done and passed.

Lin

On Tue, Jul 13, 2021 at 7:50 PM WenMeng-NOAA ***@***.***>
wrote:

> @hsinmulin-NOAA <https://github.com/hsinmulin-NOAA> Please open a new
> issue for this PR. Did you conduct the UPP tests?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EMC_post/pull/346#issuecomment-879477622>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALRSTCO77I5WOWRQ6JZS7DTTXTGMHANCNFSM5AJ7GB2A>
> .
>

WenMeng-NOAA commented on parm/fv3lam_cmaq_post_avblflds.xml:
```diff
@@ -0,0 +1,6147 @@
+<?xml version="1.0"?>
```
This file should removed and all new variables for AQ are added in post_avblflds.xml.

hsinmulin-NOAA replied:
The additional variables in this additional file and the other extra file (*cmaq*) as in the new makefile are consistent with the  "*lam*".  They are used for not put dummy variables in other domains.

WenMeng-NOAA replied:
The UPP should have an unified avblflds.xml. Please remove all new AQF definitions into post_avblflds.xml.

hsinmulin-NOAA replied:
file removed.  An unified "post_avblflds.xml" is used instead

WenMeng-NOAA commented:
@hsinmulin-NOAA Do you have the sample the model history files including these new AQF fields for the UPP standalone test?

WenMeng-NOAA commented:
@JianpingHuang-NOAA and @hsinmulin-NOAA Is there a plan to add these new AQF fields in RRFS parallel run?

hsinmulin-NOAA commented:
The fields only can be done with CMAQ involved (RRFS-CMAQ).  They can't be
processed only by RRFS

Lin

On Wed, Jul 14, 2021 at 10:23 AM WenMeng-NOAA ***@***.***>
wrote:

> @JianpingHuang-NOAA <https://github.com/JianpingHuang-NOAA> and
> @hsinmulin-NOAA <https://github.com/hsinmulin-NOAA> Is there a plan to
> add these new AQF fields in RRFS parallel run?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EMC_post/pull/346#issuecomment-879937026>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALRSTCIPZGY4UEJ6LQLTL7LTXWMWPANCNFSM5AJ7GB2A>
> .
>

hsinmulin-NOAA commented:
Those fields will be dummy variables in RRFS model output.  They only exist
in RRFS-CMAQ

On Wed, Jul 14, 2021 at 10:29 AM Hsin-Mu Lin - NOAA Affiliate <
***@***.***> wrote:

> The fields only can be done with CMAQ involved (RRFS-CMAQ).  They can't be
> processed only by RRFS
>
> Lin
>
> On Wed, Jul 14, 2021 at 10:23 AM WenMeng-NOAA ***@***.***>
> wrote:
>
>> @JianpingHuang-NOAA <https://github.com/JianpingHuang-NOAA> and
>> @hsinmulin-NOAA <https://github.com/hsinmulin-NOAA> Is there a plan to
>> add these new AQF fields in RRFS parallel run?
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/NOAA-EMC/EMC_post/pull/346#issuecomment-879937026>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/ALRSTCIPZGY4UEJ6LQLTL7LTXWMWPANCNFSM5AJ7GB2A>
>> .
>>
>

hsinmulin-NOAA commented:
Per additional new baseline of RRFS-CMAQ for the regression test is essential and the sample model out as well as UPP results are ready

WenMeng-NOAA commented:
@hsinmulin-NOAA Please provide the sample model history files for our UPP standalone test. A new issue is needed for your pr at
https://github.com/NOAA-EMC/EMC_post/issues

hsinmulin-NOAA committed:
unified "post_avblflds.xml" related change

Update files according to the removal of "fv3lam_cmaq_post_avblflds.xml" to a unified "post_avblflds.xml"

hsinmulin-NOAA committed:
Delete fv3lam_cmaq_post_avblflds.xml

to an unified "post_avblflds.xml"

hsinmulin-NOAA commented:
fv3lam_cmaq_post_avblflds.xml is removed

3 related files also updated
makefile
post_avblflds.xml
postxconfig-NT-fv3lam_cmaq



On Wed, Jul 14, 2021 at 1:31 PM WenMeng-NOAA ***@***.***>
wrote:

> *@WenMeng-NOAA* commented on this pull request.
> ------------------------------
>
> In parm/fv3lam_cmaq_post_avblflds.xml
> <https://github.com/NOAA-EMC/EMC_post/pull/346#discussion_r669815895>:
>
> > @@ -0,0 +1,6147 @@
> +<?xml version="1.0"?>
>
> The UPP should have an unified avblflds.xml. Please remove all new AQF
> definitions into post_avblflds.xml.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EMC_post/pull/346#discussion_r669815895>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALRSTCKZXVMFX46LCZ7IF7TTXXCWBANCNFSM5AJ7GB2A>
> .
>

hsinmulin-NOAA commented:
A sample model data as input can be found at
/gpfs/dell2/emc/modeling/save/Hsin-Mu.Lin/RRFS-CMAQ_output/2019081512

The UPP result is at
/gpfs/dell2/emc/modeling/noscrub/Hsin-Mu.Lin/post_fv3_2019081512

On Wed, Jul 14, 2021 at 1:34 PM WenMeng-NOAA ***@***.***>
wrote:

> @hsinmulin-NOAA <https://github.com/hsinmulin-NOAA> Please provide the
> sample model history files for our UPP standalone test. A new issue is
> needed for your pr at
> https://github.com/NOAA-EMC/EMC_post/issues
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EMC_post/pull/346#issuecomment-880080339>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALRSTCLYKCKYJP3RAACVUQDTXXDCDANCNFSM5AJ7GB2A>
> .
>

WenMeng-NOAA commented on sorc/ncep_post.fd/MDLFLD.f:
```diff
@@ -2180,6 +2183,96 @@ SUBROUTINE MDLFLD
+
+        !---- O3 ----
+
+           IF (IGET(267)>0) THEN
```
@hsinmulin-NOAA The ID 267 for output O3 is already available in this routine.  May I know the reason for adding in AQF session?

WenMeng-NOAA replied:
@hsinmulin-NOAA I would think you don't need another block of "IF(IGET(267)>0) THEN" since this block is available in the current UPP.

WenMeng-NOAA replied:
@hsinmulin-NOAA In UPP code read "o3mr"(ozone mixing) from fv3 based model history files in netcdf format, 
 assign value to an UPP interval variable "o3" and write out in GRIB2 with name "O3MR". So variable o3 is just an interval variable.

WenMeng-NOAA replied:
@hsinmulin-NOAA There is no reason you read the same variable twice in the read interface.

WenMeng-NOAA commented on parm/post_avblflds.xml:
```diff
@@ -7709,6 +7709,469 @@
        <scale>6.0</scale>
     </param>
 
+    <param>
```
@hsinmulin-NOAA May I ask why you assign the new field ID starting at 1185 instead 994?

WenMeng-NOAA commented on parm/post_avblflds.xml:
```diff
@@ -7709,6 +7709,469 @@
     </param>
 
+    <param>
+       <post_avblfldidx>1019</post_avblfldidx>
```
@hsinmulin-NOAA I can't find grib2 names of new fields you are proposing in the grib2 table https://www.nco.ncep.noaa.gov/pmb/docs/grib2/grib2_doc/grib2_table4-2-0-13.shtml. Any grib2 tables I should look for?

hsinmulin-NOAA commented:
Duplication code for previous debugging and can be eliminated

On Tue, Jul 27, 2021 at 1:04 PM WenMeng-NOAA ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In sorc/ncep_post.fd/MDLFLD.f
> <https://github.com/NOAA-EMC/EMC_post/pull/346#discussion_r677640442>:
>
> > @@ -2180,6 +2183,96 @@ SUBROUTINE MDLFLD
>                 endif
>              END IF
>             ENDIF
> +
> +!===============
> +! AQF
> +!===============
> +
> +        !---- O3 ----
> +
> +           IF (IGET(267)>0) THEN
>
> @hsinmulin-NOAA <https://github.com/hsinmulin-NOAA> The ID 267 for output
> O3 is already available in this routine. May I know the reason for adding
> in AQF session?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EMC_post/pull/346#pullrequestreview-716169548>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALRSTCP253VWQY2FL6NDWQDTZ3RKDANCNFSM5AJ7GB2A>
> .
>

hsinmulin-NOAA commented:
We have found all the available fields in the GRIB2 table at this moment.
Additional variables will be added if needed in the future

On Tue, Jul 27, 2021 at 1:15 PM WenMeng-NOAA ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In parm/post_avblflds.xml
> <https://github.com/NOAA-EMC/EMC_post/pull/346#discussion_r677648230>:
>
> > @@ -7709,6 +7709,469 @@
>         <scale>6.0</scale>
>      </param>
>
> +    <param>
> +       <post_avblfldidx>1019</post_avblfldidx>
>
> @hsinmulin-NOAA <https://github.com/hsinmulin-NOAA> I can't find grib2
> names of new fields you are proposing in the grib2 table
> https://www.nco.ncep.noaa.gov/pmb/docs/grib2/grib2_doc/grib2_table4-2-0-13.shtml.
> Any grib2 tables I should look for?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EMC_post/pull/346#pullrequestreview-716179791>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALRSTCJA5FU6ALM6I2MKJXDTZ3SUFANCNFSM5AJ7GB2A>
> .
>

WenMeng-NOAA commented:
> We have found all the available fields in the GRIB2 table at this moment. Additional variables will be added if needed in the future
> […](#)
> On Tue, Jul 27, 2021 at 1:15 PM WenMeng-NOAA ***@***.***> wrote: ***@***.**** commented on this pull request. ------------------------------ In parm/post_avblflds.xml <[#346 (comment)](https://github.com/NOAA-EMC/EMC_post/pull/346#discussion_r677648230)>: > @@ -7709,6 +7709,469 @@ <scale>6.0</scale> </param> + <param> + <post_avblfldidx>1019</post_avblfldidx> @hsinmulin-NOAA <https://github.com/hsinmulin-NOAA> I can't find grib2 names of new fields you are proposing in the grib2 table https://www.nco.ncep.noaa.gov/pmb/docs/grib2/grib2_doc/grib2_table4-2-0-13.shtml. Any grib2 tables I should look for? — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub <[#346 (review)](https://github.com/NOAA-EMC/EMC_post/pull/346#pullrequestreview-716179791)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ALRSTCJA5FU6ALM6I2MKJXDTZ3SUFANCNFSM5AJ7GB2A> .

Please add the fields which can  currently be  outputted from the model only. You are proposing 176 new 3D fields but most of them are not assigned grib2 names at https://www.nco.ncep.noaa.gov/pmb/docs/grib2/grib2_doc/grib2_table4-1.shtml and haven't incorporated the changes in UPP code to write out in GRIB2.

hsinmulin-NOAA commented:
After examining the code in detail, "O3MR" is the current official GRIB2
table
<https://www.nco.ncep.noaa.gov/pmb/docs/grib2/grib2_doc/grib2_table4-2-0-14.shtml>
variable and  is used as it is consistent with the other AQF code.
The old O3 processes before AQF code are critical and need to be examined
by the original coder.

On Tue, Jul 27, 2021 at 1:30 PM Hsin-Mu Lin - NOAA Affiliate <
***@***.***> wrote:

> Duplication code for previous debugging and can be eliminated
>
> On Tue, Jul 27, 2021 at 1:04 PM WenMeng-NOAA ***@***.***>
> wrote:
>
>> ***@***.**** commented on this pull request.
>> ------------------------------
>>
>> In sorc/ncep_post.fd/MDLFLD.f
>> <https://github.com/NOAA-EMC/EMC_post/pull/346#discussion_r677640442>:
>>
>> > @@ -2180,6 +2183,96 @@ SUBROUTINE MDLFLD
>>                 endif
>>              END IF
>>             ENDIF
>> +
>> +!===============
>> +! AQF
>> +!===============
>> +
>> +        !---- O3 ----
>> +
>> +           IF (IGET(267)>0) THEN
>>
>> @hsinmulin-NOAA <https://github.com/hsinmulin-NOAA> The ID 267 for
>> output O3 is already available in this routine. May I know the reason for
>> adding in AQF session?
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/NOAA-EMC/EMC_post/pull/346#pullrequestreview-716169548>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/ALRSTCP253VWQY2FL6NDWQDTZ3RKDANCNFSM5AJ7GB2A>
>> .
>>
>

JianpingHuang-NOAA reviewed (changes requested):
1) It is found that the grib2 files include two species, OZCON and PMTF, what about other chemical species?  We need to convert other chemical species like NO, NO2, ISOP, and other VOCs into grib2. In addition, we need PM2.5 compositions like NH4, NO3, and SOC, SOM.

2) for each NATLEV file, why do we have two lines of O3MR for each hybrid level?

hsinmulin-NOAA commented:
1. OZCON & PMTF are the only species that need to be calculated in the
UPP.  All the other species need not to be calculated and can be found in
the original netcdf file.  This is the original plan by the managers to
process the species that need to be recalculated.
2.  The O3MR is duplicated output for verification previously and can be
eliminated


On Tue, Jul 27, 2021 at 3:10 PM JianpingHuang-NOAA ***@***.***>
wrote:

> ***@***.**** requested changes on this pull request.
>
>    1.
>
>    It is found that the grib2 files include two species, OZCON and PMTF,
>    what about other chemical species? We need to convert other chemical
>    species like NO, NO2, ISOP, and other VOCs into grib2. In addition, we need
>    PM2.5 compositions like NH4, NO3, and SOC, SOM.
>    2.
>
>    for each NATLEV file, why do we have two lines of O3MR for each hybrid
>    level?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EMC_post/pull/346#pullrequestreview-716283869>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALRSTCLRTPN2KQKKU3QPPIDTZ4ABNANCNFSM5AJ7GB2A>
> .
>

hsinmulin-NOAA commented:
Yes, I agree.  The old one can be removed and group all the items under AQF

On Tue, Jul 27, 2021 at 3:38 PM WenMeng-NOAA ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In sorc/ncep_post.fd/MDLFLD.f
> <https://github.com/NOAA-EMC/EMC_post/pull/346#discussion_r677744786>:
>
> > @@ -2180,6 +2183,96 @@ SUBROUTINE MDLFLD
>                 endif
>              END IF
>             ENDIF
> +
> +!===============
> +! AQF
> +!===============
> +
> +        !---- O3 ----
> +
> +           IF (IGET(267)>0) THEN
>
> @hsinmulin-NOAA <https://github.com/hsinmulin-NOAA> I would think you
> don't need another block of "IF(IGET(267)>0) THEN" since this block is
> available in the current UPP.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EMC_post/pull/346#discussion_r677744786>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALRSTCOP54SGOECNSC6IXQDTZ4DKRANCNFSM5AJ7GB2A>
> .
>

hsinmulin-NOAA commented:
As for the 2.
It is caused by the original "params_grib2_tbl_new" that list
   0  14     1   0 O3MR
   0  14   192   1 O3MR

They are duplicated and one needs to be removed by the UPP manger.


On Tue, Jul 27, 2021 at 3:10 PM JianpingHuang-NOAA ***@***.***>
wrote:

> ***@***.**** requested changes on this pull request.
>
>    1.
>
>    It is found that the grib2 files include two species, OZCON and PMTF,
>    what about other chemical species? We need to convert other chemical
>    species like NO, NO2, ISOP, and other VOCs into grib2. In addition, we need
>    PM2.5 compositions like NH4, NO3, and SOC, SOM.
>    2.
>
>    for each NATLEV file, why do we have two lines of O3MR for each hybrid
>    level?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EMC_post/pull/346#pullrequestreview-716283869>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALRSTCLRTPN2KQKKU3QPPIDTZ4ABNANCNFSM5AJ7GB2A>
> .
>

WenMeng-NOAA commented:
That's Okay. One is for WMO official one, one is for NCEP local use.
Eventually we will remove the fields
which have WMO official names. Right now, you just leave as it is.

Thanks,

Wen

On Tue, Jul 27, 2021 at 3:53 PM hsinmulin-NOAA ***@***.***>
wrote:

> As for the 2.
> It is caused by the original "params_grib2_tbl_new" that list
> 0 14 1 0 O3MR
> 0 14 192 1 O3MR
>
> They are duplicated and one needs to be removed by the UPP manger.
>
>
> On Tue, Jul 27, 2021 at 3:10 PM JianpingHuang-NOAA ***@***.***>
> wrote:
>
> > ***@***.**** requested changes on this pull request.
> >
> > 1.
> >
> > It is found that the grib2 files include two species, OZCON and PMTF,
> > what about other chemical species? We need to convert other chemical
> > species like NO, NO2, ISOP, and other VOCs into grib2. In addition, we
> need
> > PM2.5 compositions like NH4, NO3, and SOC, SOM.
> > 2.
> >
> > for each NATLEV file, why do we have two lines of O3MR for each hybrid
> > level?
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub
> > <
> https://github.com/NOAA-EMC/EMC_post/pull/346#pullrequestreview-716283869
> >,
> > or unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/ALRSTCLRTPN2KQKKU3QPPIDTZ4ABNANCNFSM5AJ7GB2A
> >
> > .
> >
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EMC_post/pull/346#issuecomment-887790774>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALQGNESBNEXU34HNGXO4YR3TZ4FCZANCNFSM5AJ7GB2A>
> .
>


-- 
Wen Meng
Physical Scientist
Verification/Post-Processing/Product-Generation Branch
NOAA/NWS/NCEP/EMC
5830 University Research Ct., Room 2070
College Park, MD 20740
***@***.***
301-683-3779

hsinmulin-NOAA committed:
remove variables do not in the GRIB2 TABLE

Update the file for variables used in the output and defined in GRIB2 TABLE

hsinmulin-NOAA committed:
Update codes to consistent with changes in parm

Remove variables that will not be used in UPP output.

hsinmulin-NOAA commented:
The O3 is probably used by other programs and not AQF.  O3MR is the
variable in AQF netcdf model output.

On Tue, Jul 27, 2021 at 3:38 PM WenMeng-NOAA ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In sorc/ncep_post.fd/MDLFLD.f
> <https://github.com/NOAA-EMC/EMC_post/pull/346#discussion_r677744786>:
>
> > @@ -2180,6 +2183,96 @@ SUBROUTINE MDLFLD
>                 endif
>              END IF
>             ENDIF
> +
> +!===============
> +! AQF
> +!===============
> +
> +        !---- O3 ----
> +
> +           IF (IGET(267)>0) THEN
>
> @hsinmulin-NOAA <https://github.com/hsinmulin-NOAA> I would think you
> don't need another block of "IF(IGET(267)>0) THEN" since this block is
> available in the current UPP.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EMC_post/pull/346#discussion_r677744786>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALRSTCOP54SGOECNSC6IXQDTZ4DKRANCNFSM5AJ7GB2A>
> .
>

hsinmulin-NOAA commented:
files have been changed according to the requirement

On Tue, Jul 27, 2021 at 1:41 PM WenMeng-NOAA ***@***.***>
wrote:

> We have found all the available fields in the GRIB2 table at this moment.
> Additional variables will be added if needed in the future
> … <#m_-3006876892929344742_>
> On Tue, Jul 27, 2021 at 1:15 PM WenMeng-NOAA *@*.*> wrote: @.** commented
> on this pull request. ------------------------------ In
> parm/post_avblflds.xml <#346 (comment)
> <https://github.com/NOAA-EMC/EMC_post/pull/346#discussion_r677648230>>: >
> @@ -7709,6 +7709,469 @@ 6.0 + + <post_avblfldidx>1019</post_avblfldidx>
> @hsinmulin-NOAA <https://github.com/hsinmulin-NOAA>
> https://github.com/hsinmulin-NOAA I can't find grib2 names of new fields
> you are proposing in the grib2 table
> https://www.nco.ncep.noaa.gov/pmb/docs/grib2/grib2_doc/grib2_table4-2-0-13.shtml.
> Any grib2 tables I should look for? — You are receiving this because you
> were mentioned. Reply to this email directly, view it on GitHub <#346
> (review)
> <https://github.com/NOAA-EMC/EMC_post/pull/346#pullrequestreview-716179791>>,
> or unsubscribe
> https://github.com/notifications/unsubscribe-auth/ALRSTCJA5FU6ALM6I2MKJXDTZ3SUFANCNFSM5AJ7GB2A
> .
>
> Please add the fields which can currently be outputted from the model
> only. You are proposing 176 new 3D fields but most of them are not assigned
> grib2 names at
> https://www.nco.ncep.noaa.gov/pmb/docs/grib2/grib2_doc/grib2_table4-1.shtml
> and haven't incorporated the changes in UPP code to write out in GRIB2.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EMC_post/pull/346#issuecomment-887704448>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALRSTCLIBW6LJVMFJ3KXQO3TZ3VVLANCNFSM5AJ7GB2A>
> .
>

hsinmulin-NOAA commented:
codes and parms have been changed according to the requirements

On Tue, Jul 27, 2021 at 1:07 PM WenMeng-NOAA ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In parm/post_avblflds.xml
> <https://github.com/NOAA-EMC/EMC_post/pull/346#discussion_r677642189>:
>
> > @@ -7709,6 +7709,469 @@
>         <scale>6.0</scale>
>      </param>
>
> +    <param>
>
> @hsinmulin-NOAA <https://github.com/hsinmulin-NOAA> May I ask why you
> assign the new field ID starting at 1185 instead 994?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EMC_post/pull/346#pullrequestreview-716171998>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALRSTCK3BZRL2TKQA3HB7QDTZ3RUHANCNFSM5AJ7GB2A>
> .
>

WenMeng-NOAA commented on sorc/ncep_post.fd/INITPOST_NETCDF.f:
```diff
@@ -863,6 +877,555 @@ SUBROUTINE INITPOST_NETCDF(ncid3d)
+       ! === It will cause problem if not use the lower case =====
+       ! *********************************************************
+
+       VarName='o3mr'
```
@hsinmulin-NOAA The 'o3mr' is already read in the current setting. Please remove your change.

hsinmulin-NOAA commented:
The o3mr is fundamentally different by the name after call read_netcdf

Fore AQF
 888        VarName='o3mr'
 889        call read_netcdf_3d_scatter(me,ncid3d,1,im,jm,jsta,jsta_2l &
 890        ,jend_2u,MPI_COMM_COMP,icnt,idsp,spval,VarName &
 891        ,lm,*o3mr*(1,jsta_2l,1))

For original (per you mentioned)
 824        call read_netcdf_3d_scatter(me,ncid3d,1,im,jm,jsta,jsta_2l &
 825        ,jend_2u,MPI_COMM_COMP,icnt,idsp,spval,recname(5) &
 826        ,lm,*o3*(1,jsta_2l,1))

Name will be changed after reading





On Fri, Jul 30, 2021 at 1:55 PM WenMeng-NOAA ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In sorc/ncep_post.fd/INITPOST_NETCDF.f
> <https://github.com/NOAA-EMC/EMC_post/pull/346#discussion_r680125543>:
>
> > @@ -863,6 +877,555 @@ SUBROUTINE INITPOST_NETCDF(ncid3d)
>            cwm(isa,jsa,l)
>        end do
>
> +!=============================
> +! For AQF Chemical species
> +!=============================
> +
> +       ! *********** VarName need to be in lower case ************
> +       ! === It will cause problem if not use the lower case =====
> +       ! *********************************************************
> +
> +       VarName='o3mr'
>
> @hsinmulin-NOAA <https://github.com/hsinmulin-NOAA> The 'o3mr' is already
> read in the current setting. Please remove your change.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EMC_post/pull/346#pullrequestreview-719339691>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALRSTCLTMERU5XZ5NZH5JVLT2LRP5ANCNFSM5AJ7GB2A>
> .
>

hsinmulin-NOAA committed:
Remove O3MR

Remove O3MR that has been processed in the original code
The removal of O3MR in both codes also eliminate the duplicated "O3MR" in UPP output

hsinmulin-NOAA commented:
2 codes changed as necessary to remove the O3MR.  The duplicated O3MR in
UPP output is also been corrected

On Fri, Jul 30, 2021 at 3:14 PM WenMeng-NOAA ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In sorc/ncep_post.fd/MDLFLD.f
> <https://github.com/NOAA-EMC/EMC_post/pull/346#discussion_r680167243>:
>
> > @@ -2180,6 +2183,96 @@ SUBROUTINE MDLFLD
>                 endif
>              END IF
>             ENDIF
> +
> +!===============
> +! AQF
> +!===============
> +
> +        !---- O3 ----
> +
> +           IF (IGET(267)>0) THEN
>
> @hsinmulin-NOAA <https://github.com/hsinmulin-NOAA> There is no reason
> you read the same variable twice in the read interface.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EMC_post/pull/346#discussion_r680167243>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALRSTCMO4WBYYUAKV7J2ZK3T2L22BANCNFSM5AJ7GB2A>
> .
>

JianpingHuang-NOAA commented:
In fact,  O3MR is a tracer in the model. We do not use this parameter.


On Fri, Jul 30, 2021 at 3:14 PM WenMeng-NOAA ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In sorc/ncep_post.fd/MDLFLD.f
> <https://github.com/NOAA-EMC/EMC_post/pull/346#discussion_r680167243>:
>
> > @@ -2180,6 +2183,96 @@ SUBROUTINE MDLFLD
>                 endif
>              END IF
>             ENDIF
> +
> +!===============
> +! AQF
> +!===============
> +
> +        !---- O3 ----
> +
> +           IF (IGET(267)>0) THEN
>
> @hsinmulin-NOAA <https://github.com/hsinmulin-NOAA> There is no reason
> you read the same variable twice in the read interface.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EMC_post/pull/346#discussion_r680167243>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANA2PI3VC4LMIGT3QBX3FBDT2L22BANCNFSM5AJ7GB2A>
> .
>

hsinmulin-NOAA committed:
Change the allocation 

Add "aqfcmaq_on" as the itag for the conditional allocation for the input variables.  Only variables in the output are set in the main allocation, the rest are set as local in the "INITPOST_NETCDF" and will only be used when "aqfcmaq_on=.true."

hsinmulin-NOAA committed:
Initialize the AQF related variables to be "0." instead of "spval"

The purpose for the initialization is for run without itag "aqfcmaq_on=.true." (default is "aqfcmaq_on=.false.").  The value of PMTF & OZCON shown in the output will be "0" instead of misled value related to the calculation from "spval".  O3MR is the tracer also in FV3GFS, the process is outside the "aqfcmaq_on" itag control.

JianpingHuang-NOAA reviewed (commented):
Is the modified UPP is able to post-process the RRFS/CMAQ model outputs at a horizontal resolution of 13 km?

WenMeng-NOAA commented:
> Is the modified UPP is able to post-process the RRFS/CMAQ model outputs at a horizontal resolution of 13 km?

The UPP doesn't do horizontal interpolation. You may use wgrib2 to generate that kind of product.

WenMeng-NOAA commented on parm/makefile:
```diff
@@ -20,6 +20,7 @@ GEFSFLATFILENAME        = postxconfig-NT-GEFS.txt
 GEFSANLFLATFILENAME     = postxconfig-NT-GEFS-ANL.txt
 GEFSF00FLATFILENAME     = postxconfig-NT-GEFS-F00.txt
 FV3LAMFLATFILENAME      = postxconfig-NT-fv3lam.txt
+FV3LAMCMAQFLATFILENAME  = postxconfig-NT-fv3lam_cmaq.txt
```
@hsinmulin-NOAA Please revert your changes in makefile.

WenMeng-NOAA commented on sorc/ncep_post.fd/ALLOCATE_ALL.f:
```diff
@@ -1,1295 +1,1312 @@
-      enddo
-! 
-      end
+!> @file
```
@hsinmulin-NOAA Please check out a fresh version of ALLOCATE_ALL.f from upstream/develop and incorporate your changes. Your current version modify the format of the file so it is hard to track your actual changes.

WenMeng-NOAA replied:
@hsinmulin-NOAA I know you only add small part of changes, Somehow, it is a long list  in this file. So people are hard to see your actual changes.

hsinmulin-NOAA commented:
The items have no changes and exist in the code.  Please check it again.

On Wed, Aug 4, 2021 at 3:43 PM WenMeng-NOAA ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In sorc/ncep_post.fd/ALLOCATE_ALL.f
> <https://github.com/NOAA-EMC/EMC_post/pull/346#discussion_r682908055>:
>
> > -! UPP_MATH MODULE DIFFERENTIAL EQUATIONS
> -      allocate(ddvdx(im,jsta_2l:jend_2u))
> -      allocate(ddudy(im,jsta_2l:jend_2u))
> -      allocate(uuavg(im,jsta_2l:jend_2u))
> -!Initialization
> -!$omp parallel do private(i,j)
> -      do j=jsta_2l,jend_2u
> -        do i=1,im
> -          ddvdx(i,j)=spval
> -          ddudy(i,j)=spval
> -          uuavg(i,j)=spval
> -        enddo
> -      enddo
> -!
> -      end
> +!> @file
>
> @hsinmulin-NOAA <https://github.com/hsinmulin-NOAA> Please check out a
> fresh version of ALLOCATE_ALL.f from upstream/develop and incorporate your
> changes. Your current version modify the format of the file so it is hard
> to track your actual changes.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EMC_post/pull/346#pullrequestreview-722671922>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALRSTCIAOMAD6APHGVVYAVDT3GJ43ANCNFSM5AJ7GB2A>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email>
> .
>

WenMeng-NOAA commented:
Please see
https://github.com/NOAA-EMC/EMC_post/pull/346/files#diff-04eaeafceb4bc1c6f5aada69f5680dc0859563216bb39cd1a43d88dba4dd883d



On Wed, Aug 4, 2021 at 4:23 PM hsinmulin-NOAA ***@***.***>
wrote:

> The items have no changes and exist in the code. Please check it again.
>
> On Wed, Aug 4, 2021 at 3:43 PM WenMeng-NOAA ***@***.***>
> wrote:
>
> > ***@***.**** commented on this pull request.
> > ------------------------------
> >
> > In sorc/ncep_post.fd/ALLOCATE_ALL.f
> > <https://github.com/NOAA-EMC/EMC_post/pull/346#discussion_r682908055>:
> >
> > > -! UPP_MATH MODULE DIFFERENTIAL EQUATIONS
> > - allocate(ddvdx(im,jsta_2l:jend_2u))
> > - allocate(ddudy(im,jsta_2l:jend_2u))
> > - allocate(uuavg(im,jsta_2l:jend_2u))
> > -!Initialization
> > -!$omp parallel do private(i,j)
> > - do j=jsta_2l,jend_2u
> > - do i=1,im
> > - ddvdx(i,j)=spval
> > - ddudy(i,j)=spval
> > - uuavg(i,j)=spval
> > - enddo
> > - enddo
> > -!
> > - end
> > +!> @file
> >
> > @hsinmulin-NOAA <https://github.com/hsinmulin-NOAA> Please check out a
> > fresh version of ALLOCATE_ALL.f from upstream/develop and incorporate
> your
> > changes. Your current version modify the format of the file so it is hard
> > to track your actual changes.
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub
> > <
> https://github.com/NOAA-EMC/EMC_post/pull/346#pullrequestreview-722671922
> >,
> > or unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/ALRSTCIAOMAD6APHGVVYAVDT3GJ43ANCNFSM5AJ7GB2A
> >
> > .
> > Triage notifications on the go with GitHub Mobile for iOS
> > <
> https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675
> >
> > or Android
> > <
> https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email
> >
> > .
> >
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EMC_post/pull/346#issuecomment-892948819>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALQGNEUHM7GQFPZZEMJG44TT3GOSVANCNFSM5AJ7GB2A>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email>
> .
>


-- 
Wen Meng
Physical Scientist
Verification/Post-Processing/Product-Generation Branch
NOAA/NWS/NCEP/EMC
5830 University Research Ct., Room 2070
College Park, MD 20740
***@***.***
301-683-3779

hsinmulin-NOAA committed:
renew file

Reload the file to eliminate the "block & spacce" (-b) difference from the code diff

hsinmulin-NOAA commented:
Code updated pre request.
The issue is from the code editor.  empty Space and block cause issue

On Wed, Aug 4, 2021 at 4:32 PM WenMeng-NOAA ***@***.***>
wrote:

> Please see
>
> https://github.com/NOAA-EMC/EMC_post/pull/346/files#diff-04eaeafceb4bc1c6f5aada69f5680dc0859563216bb39cd1a43d88dba4dd883d
>
>
>
> On Wed, Aug 4, 2021 at 4:23 PM hsinmulin-NOAA ***@***.***>
> wrote:
>
> > The items have no changes and exist in the code. Please check it again.
> >
> > On Wed, Aug 4, 2021 at 3:43 PM WenMeng-NOAA ***@***.***>
> > wrote:
> >
> > > ***@***.**** commented on this pull request.
> > > ------------------------------
> > >
> > > In sorc/ncep_post.fd/ALLOCATE_ALL.f
> > > <https://github.com/NOAA-EMC/EMC_post/pull/346#discussion_r682908055>:
> > >
> > > > -! UPP_MATH MODULE DIFFERENTIAL EQUATIONS
> > > - allocate(ddvdx(im,jsta_2l:jend_2u))
> > > - allocate(ddudy(im,jsta_2l:jend_2u))
> > > - allocate(uuavg(im,jsta_2l:jend_2u))
> > > -!Initialization
> > > -!$omp parallel do private(i,j)
> > > - do j=jsta_2l,jend_2u
> > > - do i=1,im
> > > - ddvdx(i,j)=spval
> > > - ddudy(i,j)=spval
> > > - uuavg(i,j)=spval
> > > - enddo
> > > - enddo
> > > -!
> > > - end
> > > +!> @file
> > >
> > > @hsinmulin-NOAA <https://github.com/hsinmulin-NOAA> Please check out a
> > > fresh version of ALLOCATE_ALL.f from upstream/develop and incorporate
> > your
> > > changes. Your current version modify the format of the file so it is
> hard
> > > to track your actual changes.
> > >
> > > —
> > > You are receiving this because you were mentioned.
> > > Reply to this email directly, view it on GitHub
> > > <
> >
> https://github.com/NOAA-EMC/EMC_post/pull/346#pullrequestreview-722671922
> > >,
> > > or unsubscribe
> > > <
> >
> https://github.com/notifications/unsubscribe-auth/ALRSTCIAOMAD6APHGVVYAVDT3GJ43ANCNFSM5AJ7GB2A
> > >
> > > .
> > > Triage notifications on the go with GitHub Mobile for iOS
> > > <
> >
> https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675
> > >
> > > or Android
> > > <
> >
> https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email
> > >
> > > .
> > >
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/NOAA-EMC/EMC_post/pull/346#issuecomment-892948819>,
> > or unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/ALQGNEUHM7GQFPZZEMJG44TT3GOSVANCNFSM5AJ7GB2A
> >
> > .
> > Triage notifications on the go with GitHub Mobile for iOS
> > <
> https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675
> >
> > or Android
> > <
> https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email
> >
> > .
> >
>
>
> --
> Wen Meng
> Physical Scientist
> Verification/Post-Processing/Product-Generation Branch
> NOAA/NWS/NCEP/EMC
> 5830 University Research Ct., Room 2070
> College Park, MD 20740
> ***@***.***
> 301-683-3779
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EMC_post/pull/346#issuecomment-892954259>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALRSTCJ543TSP3FZBLQ2T33T3GPULANCNFSM5AJ7GB2A>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email>
> .
>

hsinmulin-NOAA committed:
Further limit the allociation of AQF UPP output

Further limit the allocation of AQF UPP output variables only exist when itag "aqfcmaq_on=.true." (default .false.).  With itag "aqfcmaq=.true." , the UPP output will contains the allocated variables (currently, OZCON, PMTF).  Without the itag (default .false.), the UPP output will return to original non-CMAQ version (no OZCON & PMTF)

WenMeng-NOAA commented on sorc/ncep_post.fd/INITPOST_NETCDF.f:
```diff
@@ -863,6 +1050,579 @@ SUBROUTINE INITPOST_NETCDF(ncid3d)
+! PM2.5 SPECIES
+!=========================
+
+    !   do l=1,lm
```
@hsinmulin-NOAA Please remove the commented out lines.

WenMeng-NOAA commented on sorc/ncep_post.fd/INITPOST_NETCDF.f:
```diff
@@ -45,7 +45,8 @@ SUBROUTINE INITPOST_NETCDF(ncid3d)
               vdiffmacce, mgdrag, cnvctvmmixing, ncnvctcfrac, cnvctumflx, cnvctdmflx,           &
               cnvctzgdrag, sconvmois, cnvctmgdrag, cnvctdetmflx, duwt, duem, dusd, dudp,        &
-              wh, qqg, ref_10cm
+              wh, qqg, ref_10cm, pmtf, o3mr, ozcon
```
@hsinmulin-NOAA The new variable o3mr is not used. Can you remove it in this routine and all related routines?

hsinmulin-NOAA commented:
Those PM25 species will be added when grib2 table and wgrib2 are
available.

On Thu, Aug 5, 2021 at 11:25 AM WenMeng-NOAA ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In sorc/ncep_post.fd/INITPOST_NETCDF.f
> <https://github.com/NOAA-EMC/EMC_post/pull/346#discussion_r683559417>:
>
> > +
> +       VarName='pm25at'
> +       call read_netcdf_3d_scatter(me,ncid3d,1,im,jm,jsta,jsta_2l &
> +       ,jend_2u,MPI_COMM_COMP,icnt,idsp,spval,VarName &
> +       ,lm,pm25at(1,jsta_2l,1))
> +
> +       VarName='pm25co'
> +       call read_netcdf_3d_scatter(me,ncid3d,1,im,jm,jsta,jsta_2l &
> +       ,jend_2u,MPI_COMM_COMP,icnt,idsp,spval,VarName &
> +       ,lm,pm25co(1,jsta_2l,1))
> +
> +!=========================
> +! PM2.5 SPECIES
> +!=========================
> +
> +    !   do l=1,lm
>
> @hsinmulin-NOAA <https://github.com/hsinmulin-NOAA> Please remove the
> commented out lines.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EMC_post/pull/346#pullrequestreview-723497619>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALRSTCO3763RPFQLBWWSNYTT3KUORANCNFSM5AJ7GB2A>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email>
> .
>

hsinmulin-NOAA committed:
Remove O3MR

Remove O3mr that has been coded in the original code as "O3" from related codes

hsinmulin-NOAA commented:
O3mr has been removed from the related codes per update

On Thu, Aug 5, 2021 at 11:27 AM WenMeng-NOAA ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In sorc/ncep_post.fd/INITPOST_NETCDF.f
> <https://github.com/NOAA-EMC/EMC_post/pull/346#discussion_r683561537>:
>
> > @@ -45,7 +45,8 @@ SUBROUTINE INITPOST_NETCDF(ncid3d)
>                o3vdiff, o3prod, o3tndy, mwpv, unknown, vdiffzacce, zgdrag,cnvctummixing,         &
>                vdiffmacce, mgdrag, cnvctvmmixing, ncnvctcfrac, cnvctumflx, cnvctdmflx,           &
>                cnvctzgdrag, sconvmois, cnvctmgdrag, cnvctdetmflx, duwt, duem, dusd, dudp,        &
> -              wh, qqg, ref_10cm
> +              wh, qqg, ref_10cm, pmtf, o3mr, ozcon
>
> @hsinmulin-NOAA <https://github.com/hsinmulin-NOAA> The new variable o3mr
> is not used. Can you remove it in this routine and all related routines?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EMC_post/pull/346#pullrequestreview-723500654>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALRSTCM5UP35SWCJNKQ6WJ3T3KUXLANCNFSM5AJ7GB2A>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email>
> .
>

hsinmulin-NOAA committed:
change makefile 

comment out the RRFS-CMAQ related process currently for it is not in the RRFS prallel run

hsinmulin-NOAA commented:
The RRFS-CMAQ related processes have been commented out

On Wed, Aug 4, 2021 at 3:38 PM WenMeng-NOAA ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In parm/makefile
> <https://github.com/NOAA-EMC/EMC_post/pull/346#discussion_r682905423>:
>
> > @@ -20,6 +20,7 @@ GEFSFLATFILENAME        = postxconfig-NT-GEFS.txt
>  GEFSANLFLATFILENAME     = postxconfig-NT-GEFS-ANL.txt
>  GEFSF00FLATFILENAME     = postxconfig-NT-GEFS-F00.txt
>  FV3LAMFLATFILENAME      = postxconfig-NT-fv3lam.txt
> +FV3LAMCMAQFLATFILENAME  = postxconfig-NT-fv3lam_cmaq.txt
>
> @hsinmulin-NOAA <https://github.com/hsinmulin-NOAA> Please revert your
> changes in makefile.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EMC_post/pull/346#pullrequestreview-722668434>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALRSTCM2ROFKS2DJKNHONUTT3GJMVANCNFSM5AJ7GB2A>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email>
> .
>

hsinmulin-NOAA committed:
New NT file for chemistry only

file contains only chemical species that have been added to the source code "initpost_netcdf".

hsinmulin-NOAA committed:
Revert makefile to original

exclude the generation of RRFS-CMAQ control file related changes until parallel setup.

hsinmulin-NOAA committed:
Merge branch 'NOAA-EMC:develop' into develop

WenMeng-NOAA commented:
The UPP regression tests were conducted on WCOSS-Dell, WCOSS_Cray and Hera. No changed results were found. This PR is ready for merging.
